cwms-data-api/src/test/java/cwms/cda/data/dao/watersupply/WaterContractDaoTestIT.java

```diff
@@ -0,0 +1,422 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+import static org.junit.jupiter.api.Assertions.*;
+
+class WaterContractDaoTestIT extends DataApiTestIT {
+    private static final String OFFICE_ID = "SPK";
+    private static final String DELETE_ACTION = "DELETE ALL";
+    private static final Location testLocation = buildTestLocation("Test Location Name", "Test Location");
+    private static final Project testProject = buildTestProject();
+    private static final WaterUser testUser = buildTestWaterUser("Test User");
+
+    @BeforeAll
+    static void setup() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            LocationsDaoImpl dao = new LocationsDaoImpl(ctx);
+            ProjectDao projectDao = new ProjectDao(ctx);
+            try {
+                projectDao.create(testProject);
+                dao.storeLocation(testLocation);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to store location or project", e);
+            }
+        }, CwmsDataApiSetupCallback.getWebUser());
+    }
+
+    @AfterAll
+    static void tearDown() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            ProjectDao projectDao = new ProjectDao(ctx);
+            LocationsDaoImpl locationDao = new LocationsDaoImpl(ctx);
+            projectDao.delete(OFFICE_ID, testProject.getLocation().getName(), DeleteRule.DELETE_ALL);
+            locationDao.deleteLocation(testLocation.getName(), testLocation.getOfficeId(), false);
+        }, CwmsDataApiSetupCallback.getWebUser());
+    }
+
+    @Test
+    void testStoreAndRetrieveWaterUserList() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(buildTestWaterUser("TEST RETRIEVE USER 1"), false);
+            dao.storeWaterUser(buildTestWaterUser("TEST RETRIEVE USER 2"), false);
+            List<WaterUser> results = dao.getAllWaterUsers(new CwmsId.Builder().withOfficeId(OFFICE_ID).withName(testLocation.getName()).build());
+            DTOMatch.assertMatch(results.get(0), buildTestWaterUser("TEST RETRIEVE USER 1"));
+            DTOMatch.assertMatch(results.get(1), buildTestWaterUser("TEST RETRIEVE USER 2"));
+        }, CwmsDataApiSetupCallback.getWebUser());
+        cleanupUserRoutine(buildTestWaterUser("TEST RETRIEVE USER 1"));
```

Comment:
cleanup will not get called if the test fails, recommend adding to the @AfterEach or @AfterAll

Reply:
Moved cleanup into AfterEach function